src/main/java/fr/zcraft/zlib/components/gui/ArrayPromptGui.java

```diff
@@ -7,87 +7,105 @@
 import fr.zcraft.zlib.components.gui.Gui;
 import fr.zcraft.zlib.tools.Callback;
 
-/* Built from recommendations in PR https://github.com/zDevelopers/zLib/pull/14 */
-public class ArrayPromptGui<T> extends ExplorerGui<T> {
+public abstract class ArrayPromptGui<T> extends ExplorerGui<T> {
 
 	private Callback<T> cb;
-	private String purpose;
-	private T[] datas;
-	private ItemRenderer<T> renderer;
+	private String title;
+	private T[] data;
 	private boolean closeOnChoice;
 
 	/**
 	 * @param callback
 	 *            Callback called when the player made a choice
 	 * @param player
 	 *            The player making the choice
-	 * @param purpose
-	 *            The purpose (gui's title)
-	 * @param datas
+	 * @param title
+	 *            The gui title
+	 * @param data
 	 *            An array of datas to display
 	 * @param renderer
 	 *            Interface for building gui's ItemStack from array values
 	 * @param closeOnChoice
 	 *            Close the interface when the player has choosen if true
 	 */
-	public <A> ArrayPromptGui(Callback<T> callback, Player player, String purpose, T[] datas, ItemRenderer<T> renderer,
-			boolean closeOnChoice) {
+	public <A> ArrayPromptGui(Callback<T> callback, Player player, String title, T[] data, boolean closeOnChoice) {
 		this.cb = callback;
-		this.purpose = purpose;
-		this.datas = datas;
-		this.renderer = renderer;
+		this.title = title;
+		this.data = data;
 		this.closeOnChoice = closeOnChoice;
 
 		Gui.open(player, this);
 	}
 
+	/**
+	 * @see #onChoice(Object)
+	 * 
+	 *      Constructor with no callback argument. Note that you must override
+	 *      the onChoice method by using this constructor
+	 * 
+	 * @param player
+	 *            The player making the choice
+	 * @param title
+	 *            The gui title
+	 * @param data
+	 *            An array of datas to display
+	 * @param renderer
+	 *            Interface for building gui's ItemStack from array values
+	 * @param closeOnChoice
+	 *            Close the interface when the player has choosen if true
+	 */
+	public <A> ArrayPromptGui(Player player, String title, T[] data, boolean closeOnChoice) {
+		this(null, player, title, data, closeOnChoice);
+	}
+
+	/**
+	 * Convert an object to an ItemStack
+	 * 
+	 * @return The ItemStack to display
+	 */
+	public abstract ItemStack getViewItem(T data);
+
+	/**
+	 * Called when player made a choice if no callback was provided
+	 * 
+	 * @param data
+	 */
+	public void onChoice(T data) {
+		System.err.println("Damn ! I'm not properly implemented : override me or use a callback.");
```

Comment:
Prefer throwing an `UnsupportedOperationException` (or maybe `org.apache.commons.lang.NotImplementedException`) for these kind of errors.